apps/src/gamelab/GameLabSprite.js

```diff
@@ -42,210 +41,220 @@ module.exports.createSprite = function (x, y, width, height) {
    */
   s._verticalStretch = 1;
 
-  s.setAnimation = function (animationName) {
-    if (animationName === s.getAnimationLabel()) {
-      return;
-    }
-
-    var animation = p5Inst.projectAnimations[animationName];
-    if (typeof animation === 'undefined') {
-      throw new Error('Unable to find an animation named "' + animationName +
-          '".  Please make sure the animation exists.');
-    }
-    s.addAnimation(animationName, animation);
-    s.changeAnimation(animationName);
-    if (level.pauseAnimationsByDefault) {
-      s.pause();
-    }
-  };
-
-  s.frameDidChange = function () {
-    return s.animation ? s.animation.frameChanged : false;
-  };
-
-  s.pointTo = function (x, y) {
-    var yDelta = y - s.position.y;
-    var xDelta = x - s.position.x;
-    if (!isNaN(xDelta) && !isNaN(yDelta) && (xDelta !== 0 || yDelta !== 0)) {
-      var radiansAngle = Math.atan2(yDelta, xDelta);
-      s.rotation = 360 * radiansAngle / (2 * Math.PI);
-    }
-  };
-
-  // The scale value should include the horizontal stretch for animations.
-  s._getScaleX = function () {
-    return s.scale * s._horizontalStretch;
-  };
-
-  // The scale value should include the vertical stretch for animations.
-  s._getScaleY = function () {
-    return s.scale * s._verticalStretch;
-  };
-
-  /*
-   * @private
-   * For game lab, don't update the animation sizes because all frames are the same size.
-   */
-  s._syncAnimationSizes = function (animations, currentAnimation) {};
-
-  Object.defineProperty(s, 'frameDelay', {
-    enumerable: true,
-    get: function () {
-      if (s.animation) {
-        return s.animation.frameDelay;
-      }
-    },
-    set: function (value) {
-      if (s.animation) {
-        s.animation.frameDelay = value;
-      }
-    }
-  });
-
-  // Overriding these allows users to set a width for
+  // Overriding these allows users to set a width and height for
   // an animated sprite the same way they would an unanimated sprite.
   Object.defineProperty(s, 'width', {
     enumerable: true,
     configurable: true,
-    get: function () {
-      if (s._internalWidth === undefined) {
-        return 100;
-      } else if (s.animation) {
-        return s._internalWidth * s._horizontalStretch;
-      } else {
-        return s._internalWidth;
-      }
-    },
-    set: function (value) {
-      if (s.animation) {
-        s._horizontalStretch = value / s._internalWidth;
-      } else {
-        s._internalWidth = value;
-      }
-    }
+    get: getWidth,
+    set: setWidth
```

Comment:
Even for these `Object.defineProperty` calls, the getters and setters are generic, not bound to a particular sprite.  We don't need to create them every time.